### ISSUE TITLE
remove deprecated prometheus annotations

### ISSUE THREAD

**Issue by moolen** [external-secrets/external-secrets#1025]
#917 deprecated the old prometheus related values with their removal scheduled to `0.7.0`.

AC:
* remove `prometheus.*.enabled` API from `values.yaml`
* remove `prometheus.*.service.port` and hard-code it instead - there should be no need to configure it
* remove `service.yaml`
* remove `prometheus.io/(path|scrape|port)` annotations

**Comment by sebagomez** [comment]
hey @moolen I have a couple of question regarding this issue. 
We want to remove pretty much every configuration regarding Prometheus, right?

And we do want maintain `metrics` and everywhere it's being referenced like `webhook` and `certController`.

And basically instead of asking if Prometheus is enabled and setting its port, we would ask if metrics is enabled and if true set its values, and else hard-code promethus values, am I right?

**Comment by moolen** [comment]
> We want to remove pretty much every configuration regarding Prometheus, right?

yes, everyhting under `prometheus.*` this has been superseded by `serviceMonitor.*`

> do we want to maintain `metrics`

I had to dig this one, see here: https://github.com/external-secrets/external-secrets/pull/1392#issuecomment-1197402949: yes, we'd like to keep it so users of datadog (or other means) can make use of it.

> And basically instead of asking if Prometheus is enabled and setting its port, we would ask if metrics is enabled and if true set its values, and else hard-code prometheus values, am I right?


Not sure if i understand you here. :thinking: The prometheus port values shouldn't be used, the `metrics.port` should be used instead.

**Comment by sebagomez** [comment]
Ok, so if `serviceMonitor` is enabled, we assume it's Prometheus port (8080) since there's no `serviceMonitor.Service.port`, correct?

Also, in the services.yaml file we currently have something like this  
```yaml
  {{- if .Values.prometheus.enabled }}
  annotations:
    prometheus.io/path: "/metrics"
    prometheus.io/scrape: "true"
    prometheus.io/port: {{ .Values.prometheus.service.port | quote }}
  {{- else }}
  {{- with .Values.metrics.service.annotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
  {{- end }}
```

so this should be changed to this   
```yaml
  {{- if .Values.serviceMonitor.enabled }}
  {{- with .Values.metrics.service.annotations }}
  annotations:
    {{- toYaml . | nindent 4 }}
  {{- end }}
  {{- end }}
```

Can both `serviceMetrics` and `metrics.service` be enabled?

**Comment by moolen** [comment]
> an both `serviceMetrics` and `metrics.service` be enabled?

I would expect no, but you never what users do :wink:

yes to your above mentioned changes.

But i think i now see your problem, we would want to make the port configurable:

https://github.com/external-secrets/external-secrets/blob/762f6dc4fce7cb0787f36e7fdd5c887704a66a0d/deploy/charts/external-secrets/templates/cert-controller-deployment.yaml#L70-L73

https://github.com/external-secrets/external-secrets/blob/762f6dc4fce7cb0787f36e7fdd5c887704a66a0d/deploy/charts/external-secrets/templates/cert-controller-deployment.yaml#L57

Yes, i guess we should pull the metrics port from the relevant `metrics.service.port`.

**Comment by sebagomez** [comment]
More questions @moolen because I understand somethings from the issue but then from the code I understand something else.

https://github.com/external-secrets/external-secrets/blob/762f6dc4fce7cb0787f36e7fdd5c887704a66a0d/deploy/charts/external-secrets/templates/service.yaml#L23-L27

Here it's assumed that if Prometheus is not enabled, metrics is. Should I continue with that pattern?

As it is somebody, could not enable neither of them, and that's the reflected in the code. 
What are your thoughts?

**Comment by moolen** [comment]
This particular service is only rendered if one of these two is enabled. As far as i can see it is only needed for the `metrics.enabled==true`.

https://github.com/external-secrets/external-secrets/blob/762f6dc4fce7cb0787f36e7fdd5c887704a66a0d/deploy/charts/external-secrets/templates/service.yaml#L1

Moving forward we need this service only if `metrics.enabled==true`.

The other related service is defined inline in `servicemonitor.yaml`.
